dist/stat_data.py

```diff
@@ -879,6 +879,32 @@ def __init__(self, name, desc, flags=''):
     CursorStat('cursor_search_near_prefix_fast_paths', 'Total number of times a search near has exited due to prefix config'),
     CursorStat('cursor_skip_hs_cur_position', 'Total number of entries skipped to position the history store cursor'),
 
+    ##########################################
+    # Cursor API error statistics
+    ##########################################
+    CursorStat('cursor_bound_error', 'cursor bound calls that return an error'),
+    CursorStat('cursor_cache_error', 'cursor cache calls that return an error'),
+    CursorStat('cursor_close_error', 'cursor close calls that return an error'),
+    CursorStat('cursor_compare_error', 'cursor compare calls that return an error'),
+    CursorStat('cursor_equals_error', 'cursor equals calls that return an error'),
+    CursorStat('cursor_get_key_error', 'cursor get key calls that return an error'),
+    CursorStat('cursor_get_value_error', 'cursor get key calls that return an error'),
+    CursorStat('cursor_insert_error', 'cursor insert calls that return an error'),
+    CursorStat('cursor_insert_check_error', 'cursor insert check calls that return an error'),
+    CursorStat('cursor_modify_error', 'cursor modify calls that return an error'),
+    CursorStat('cursor_largest_key_error', 'cursor largest key calls that return an error'),
+    CursorStat('cursor_next_error', 'cursor next calls that return an error'),
+    CursorStat('cursor_next_random_error', 'cursor random next calls that return an error'),
+    CursorStat('cursor_prev_error', 'cursor prev calls that return an error'),
+    CursorStat('cursor_reconfigure_error', 'cursor reconfigure calls that return an error'),
+    CursorStat('cursor_reset_error', 'cursor reset calls that return an error'),
+    CursorStat('cursor_reserve_error', 'cursor reserve calls that return an error'),
+    CursorStat('cursor_reopen_error', 'cursor reopen calls that return an error'),
+    CursorStat('cursor_remove_error', 'cursor remove calls that return an error'),
+    CursorStat('cursor_search_near_error', 'cursor search near calls that return an error'),
+    CursorStat('cursor_search_error', 'cursor search calls that return an error'),
+    CursorStat('cursor_update_error', 'cursor update calls that return an error'),
+
```

Comment:
Set_key and set_value both have `void` return types, which is why they do not have associated statistics. The underlying `set_keyv` and `set_valuev` could in theory increment a statistic but it is a bit odd as the error code wouldn't be returned to the user.

Reply:
> it is a bit odd as the error code wouldn't be returned to the user.

This is going solely off reading the docs so I might be misunderstanding it, but won't it get returned to the user on the next cursor API call that returns an error?

It's fine as-is though, the only reason I mentioned it was that it could be confusing to see a spike in errors from (say) `search` when really it was setting keys/values that was failing.

Reply:
I don't think it would get returned to the user as part of the new API call but I could be wrong, where did you see that in the docs?

Reply:
I was looking at https://source.wiredtiger.com/develop/struct_w_t___c_u_r_s_o_r.html#ad1088d719df40babc1f57d086691ebdc, specifically the text "If an error occurs during this operation, a flag will be set in the cursor, and the next operation to access the key will fail."

Reply:
I think for the purposes of this statistic we'd only want to increment it if the API returned non-zero, which isn't really the case there. But it makes sense that you can't access a key if you failed to set it.